modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordProjectService.kt

```diff
@@ -0,0 +1,86 @@
+package com.metalbear.mirrord
+
+import com.intellij.ide.BrowserUtil
+import com.intellij.notification.Notification
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.VirtualFileManager
+
+@Service(Service.Level.PROJECT)
+class MirrordProjectService(val project: Project) : Disposable {
+    val configApi: MirrordConfigAPI = MirrordConfigAPI(this)
+
+    val execManager: MirrordExecManager = MirrordExecManager(this)
+
+    val versionCheck: MirrordVersionCheck = MirrordVersionCheck(this)
+
+    val binaryManager: MirrordBinaryManager = MirrordBinaryManager(this)
+
+    val mirrordApi: MirrordApi = MirrordApi(this)
+
+    val notifier: MirrordNotifier = MirrordNotifier(this)
+
+    @Volatile
+    var activeConfig: VirtualFile? = null
+
+    @Volatile
+    private var _enabled = false
+
+    var enabled: Boolean
+        get() = _enabled
+        set(value) {
+            if (value == _enabled) {
+                return
+            }
+
+            if (value) {
+                notifier.notifySimple("mirrord enabled", NotificationType.INFORMATION)
+            } else {
+                notifier.notifySimple("mirrord disabled", NotificationType.INFORMATION)
+            }
+
+            this._enabled = value
+
+            if (MirrordSettingsState.instance.mirrordState.telemetryEnabled == null) {
+                telemetryConsent()
+            }
+        }
+
+    /**
+     * Shows a notification asking for consent to send telemetries
+     */
+    private fun telemetryConsent() {
+        ApplicationManager.getApplication().invokeLater {
+            notifier
+                .notification(
+                    "Allow mirrord to send telemetries",
```

Comment:
I think we should rephrase it to "Allow mirrord to check for plugin updates" and leave only `versionCheckEnabled`

Reply:
So always send telemetries?